src/inc/msquic.h

```diff
@@ -817,6 +820,9 @@ typedef struct QUIC_CONNECTION_EVENT {
             uint32_t ResumptionTicketLength;
             const uint8_t* ResumptionTicket;
         } RESUMPTION_TICKET_RECEIVED;
+        struct {
+            void* PeerCertificate; // TODO - Type
```

Comment:
@scalablecory, @bartonjs, @anrossi, @ThadHouse I need some help figuring out exactly how we should indicate the "peer's certificate" up to the app here. I can think of a couple possible options:

1. Byte buffer, wire-format certificate chain.
2. Platform independent structure with functions to abstract working with the platform-specific certificate object.
3. A void* to an actual platform specific object, therefore requiring the app to already "know" the type.

1 seems the cleanest approach in my opinion, but might require the most work on the app. Thoughts?

Reply:
An additional point of complexity: With OpenSSL, we get a callback from OpenSSL with the certificate context/pointer and as far as I can tell, it only lives for the lifetime of the callback. So, if we want to support the app being able to return PENDING from this (which I think we **really** do) we can't really return OpenSSL's object without somehow copying it or the data it represents.

Reply:
Assuming that all that this layer wants to do is say "I got a certificate here, do what you want to do with it?", pbData/cbData seems the best to me.

Reply:
Just to be clear @bartonjs you're vote is for (1)?

Reply:
> Just to be clear @bartonjs you're vote is for (1)?

Yep.  The wire format is almost certainly the "raw"/DER-encoded X.509 public key certificate.  If there's some protocol-specific structure around it I'd skip that part and make the X.509 DER data pointer be what gets shared out.

Reply:
I prefer (1) as well. Yes, there's a little more memory usage and complexity, but any app  that wants to work with a certificate will be able to handle the plain DER/ASN.1 certificate bytes.

@nibanks, you mention the whole chain, however. Do you want to split out the leaf certificate and present the chain separately? Or just one whole blob, in the TLS format (3byte-length,certificate data,...)?

Reply:
I think I prefer (1) as well, except in the case of non-cross-platform clients (http.sys). If Schannel gives us back a `PCERT_CONTEXT` and then we have to convert it to the raw encoding, just for HTTP to convert it back, that seems a bit silly.

And to your question @anrossi I'd return the whole chain in one blob like TLS.